ui/web-v2/apps/admin/src/components/FeatureTargetingForm/index.tsx

```diff
@@ -69,7 +71,17 @@ export const FeatureTargetingForm: FC<FeatureTargetingFormProps> = memo(
         label: createVariationLabel(v),
       };
     });
-    const isValid = Object.keys(errors).length == 0;
+
+    const checkSaveBtnDisabled = useCallback(() => {
```

Comment:
@bimalgrg519 
Thank you for your PR!

I found a few issues. Can you take a look, please?

- When we select `User segment` and the segment from the menu [1], the `Save with comment` keeps disabled
- We have 2 conditions [2] in the rule. If you delete the first one and add a new one, it glitches the values.

[1]

![Screen Shot 2023-01-10 at 11 53 10](https://user-images.githubusercontent.com/2486691/211451846-27170e44-36b2-45ba-845c-33ff3c175352.png)

[2]

#### How to reproduce
1- Add a `Compare` condition and set its values
2- Add a `Date` condition and set its values
3- Delete the `Compare` condition
4- Click to add a new condition

![Screen Shot 2023-01-10 at 12 03 56](https://user-images.githubusercontent.com/2486691/211452525-b9c17826-8718-46dc-8805-c56c91076e24.png)